status-im/status-react
# Implement network settings for wallet settings

[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

fixes #19924 
fixes #19925 

### Summary

* This PR attempts to implement the initial network settings section of the wallet settings
  * The network settings will display the currently supported layer-1 and layer-2 networks, but without access to those network settings.
* This PR attempts to implement the initial advanced settings for toggling the Testnet mode from inside the wallet settings.

### Testing notes

* If testing this PR locally, then make sure that `ENABLE_ALERT_BANNER` is set to `1` in the `.env` file.
* If testing this PR on nightly, then `ENABLE_ALERT_BANNER` should already be activated.

#### Platforms


- Android
- iOS

#### Areas that maybe impacted


##### Functional

- Wallet Settings
- Wallet Settings - Network Settings

### Steps to test


- Open Status mobile app
- Open the settings screen
- Ensure the `wallet-settings` and `network-settings` feature flags are active
- Navigate to the wallet settings screen
- Navigate to the network settings screen
- Press the Testnet mode toggle under advanced settings
- Repeat these steps to test the Testnet mode toggle again.



### Before and after screenshots comparison

### After changes

| iOS | Android
|---|---|
| <video src="https://github.com/status-im/status-mobile/assets/2845768/0f08f981-6028-4d6d-93b0-945a633b1f5d" /> | <video src="https://github.com/status-im/status-mobile/assets/2845768/a5dc48f0-3ef8-4747-8621-541532e35b50" /> |

status: ready
